### PR TITLE
fix defects reported by Coverity Scan

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libsystemd/sd-bus/sd-bus.c [hunk 1]
```diff
@@ -3102,10 +3102,8 @@ static int add_match_callback(
                 r = 1;
         }
 
-        if (failed && match_slot->floating) {
+        if (failed && match_slot->floating)
                 bus_slot_disconnect(match_slot);
-                sd_bus_slot_unref(match_slot);
```
Comment (poettering) [inline src/libsystemd/sd-bus/sd-bus.c]: this fix isn't right. Coverity is wrong here... when the slot is "floating" it has an extra ref, and we need to get rid of it here...

(The concept of "floating" means, the "sd_bus_slot" ptr was NULL when the match was registered and hence the matche's lifetime is bound to the lifetime of the sd_bus object it belongs to. In that case we keep an extra ref on it, which makes sure the match survives as long as needed)

Comment (keszybz) [reply src/libsystemd/sd-bus/sd-bus.c]: Yeah, I was just looking at that, and it seems that the sd_bus_slot unref at the end matches the ref at the beginning, and no other ref changes occur in between.

Comment (yuwata) [reply src/libsystemd/sd-bus/sd-bus.c]: Oh, indeed. I will drop this. Thanks.

**Comment by yuwata** [comment]
@poettering @keszybz Updated.

**Pull request merged**